kani-driver/src/concrete_playback.rs

```diff
@@ -82,88 +83,48 @@ impl KaniSession {
     /// Add the unit test to the user's source code, format it, and short circuit if code already present.
     fn modify_src_code(
         &self,
-        src_path: &str,
+        src_path_as_str: &str,
         proof_harness_end_line: usize,
         concrete_playback: &UnitTest,
     ) -> Result<()> {
-        let mut src_file = File::open(src_path)
-            .with_context(|| format!("Couldn't open user's source code file `{src_path}`"))?;
-        let mut src_as_str = String::new();
-        src_file.read_to_string(&mut src_as_str).with_context(|| {
-            format!("Couldn't read user's source code file `{src_path}` as a string")
-        })?;
-
-        // Short circuit if unit test already in source code.
-        if src_as_str.contains(&concrete_playback.unit_test_name) {
+        let tmp_src_path = src_path_as_str.to_string() + ".concrete_playback_overwrite";
+        let unit_test_already_in_src = write_new_src_to_tmp_file(
+            src_path_as_str,
+            &tmp_src_path,
+            proof_harness_end_line,
+            concrete_playback,
+        )?;
```

Comment:
I don't think you can use `?` here since you still need to remove the temporary file.